pawno/include/modules/src/spectate.inc

```diff
@@ -140,11 +140,18 @@ UpdatePlayerSpectators(playerid)
 	else
 	{
 		new str[SPECTATORS_STRING_SIZE];
-		format(str, sizeof str, "%sWATCHING~n~~n~", MAIN_TEXT_COLOUR);
+		new totalAdded;
+		
+		format(str, sizeof str, "%sSPECTATORS~n~~n~", MAIN_TEXT_COLOUR);
 		foreach(new i : PlayerSpectators[playerid]) // Loop through the array that contains this player's spectators
 		{
+		    // Make sure we don't show more players than MAX_PLAYER_SPECTATORS. Otherwise string will go beyond 255th character and cause crashes.
+		    if(totalAdded >= MAX_PLAYER_SPECTATORS)break;
+		
 			// Format string with spectator name
-			format(str, sizeof str, "%s%s~n~", str, Player[i][NameWithoutTag]);
+			format(str, sizeof str, "%s%s ~r~~h~%d~n~", str, Player[i][NameWithoutTag], i);
+			
+			totalAdded++;
```

Comment:
Your limit didn't let people spectate others because limit was reached (Wtf is that?) Why I can't spectate other people? The only reason why limit is needed is the textdraw that I took care of.